src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationSource.cs

```diff
@@ -32,7 +39,19 @@ public class DaprSecretStoreConfigurationSource : IConfigurationSource
         /// <inheritdoc />
         public IConfigurationProvider Build(IConfigurationBuilder builder)
         {
-            return new DaprSecretStoreConfigurationProvider(Store, SecretDescriptors, Client);
+            if (SecretDescriptors != null)
+            {
+                if (Metadata != null)
+                { 
+                    throw new ArgumentException($"{nameof(Metadata)} must be null when {nameof(SecretDescriptors)} is set", nameof(Metadata));
```

Comment:
Is this really true? I don't think these are exclusive in the Dapr API

Reply:
`Metadata` parameter is only useful when using the bulk secret API, this means that if descriptors (Each one holds it's own metadata) are specified the parameter is not used.

That's the reason behind the 2 constructors in `DaprSecretStoreConfigurationProvider`.

Reply:
I guess what I'm really asking is this. **Both** the get secret and get bulk secret APIs in Dapr support metadata
- https://github.com/dapr/dapr/blob/master/dapr/proto/runtime/v1/dapr.proto#L238
- https://github.com/dapr/dapr/blob/master/dapr/proto/runtime/v1/dapr.proto#L257

So why are we modeling the .NET SDK around a different feature set - it seems like we should support metadata for both kinds of requests.

Reply:
Not sure I understand your question. It does allow for metadata to be used in both kind of requests. When using descriptors the api allow to specify metadata for each  requested secret, via it´s descriptor) since the underlying API is GetSecrets. Then if you just want all the secrets (using GetBulSecrets) you can specify the metadata for that call if needed.

Perhaps the confusion is that current implementation  allows the use of both APIs depending on the parameters.

Reply:
OK makes sense 👍